packages/terser/src/index.js

```diff
@@ -22,48 +22,65 @@ function log_error(...m) {
   console.error('[terser/index.js]', ...m);
 }
 
-// Peek at the arguments to find any directories declared as inputs
-let argv = process.argv.slice(2);
-// terser_minified.bzl always passes the inputs first,
-// then --output [out], then remaining args
-// We want to keep those remaining ones to pass to terser
-// Avoid a dependency on a library like minimist; keep it simple.
-const outputArgIndex = argv.findIndex((arg) => arg.startsWith('--'));
-
-// We don't want to implement a command-line parser for terser
-// so we invoke its CLI as child processes when a directory is provided, just altering the
-// input/output arguments. See discussion: https://github.com/bazelbuild/rules_nodejs/issues/822
-
-const inputs = argv.slice(0, outputArgIndex);
-const output = argv[outputArgIndex + 1];
-const residual = argv.slice(outputArgIndex + 2);
-
-// user override for the terser binary location. used in testing.
-const TERSER_BINARY = process.env.TERSER_BINARY || require.resolve('terser/bin/uglifyjs')
-// choose a default concurrency of the number of cores -1 but at least 1.
-const TERSER_CONCURENCY = (process.env.TERSER_CONCURRENCY || os.cpus().length - 1) || 1
-
-log_verbose(`Running terser/index.js
-  inputs: ${inputs}
-  output: ${output}
-  residual: ${residual}`);
+/**
+ * @param {string[]} residualArgs
+ * @param {string} output
+ */
+function processSourceMapArg(residualArgs, output) {
```

Comment:
a better name so that it is clear that this is for the directory case only would be better I think (maybe just `directoryArgs()`. I would also define this function within terserDirectory as well next to exec() so that its clear what the scope is.

Reply:
`directoryArgs()` will also be a good place to check if `${inputFile}.map` exists and if it does then `content=inline` needs to be replaced with `content='${inputFile}.map'`. That can be a follow up PR.

Reply:
we cant define it in `terserDirectory ` unforunately since we need to export it to unit test it

Reply:
ahhh. yes. that makes sense. the name change is clear enough then :)